Feat/survey management

Ray276341 opened the pull request.

giolynx104 reviewed (commented):
I've reviewed the code and functionalities. Here are a few points for discussion:

- There’s no **Submit** button on the overdue page of the submit screen. Should we consider implementing a "turn in late" feature?
- I'll make an edit to apply the **palette colors** consistently across the UI.
- The **usability** of the submit survey screen could be improved. As a new user, it’s a bit unclear what's happening. I suggest adding more descriptive labels and instructions for the fields and submission process.
- Why is there an **exclamation mark** after the submission date? Could you clarify its purpose? IMO, the text style is a bit inconsistent being black color instead of gray color like the other fields
- Should we use a button to open the survey? IMO, probably the users could click the survey directly.
Let me know your thoughts on these points, and feel free to share any suggestions for improvement!

Ray276341 commented:
> I've reviewed the code and functionalities. Here are a few points for discussion:
> 
> * There’s no **Submit** button on the overdue page of the submit screen. Should we consider implementing a "turn in late" feature?
> * I'll make an edit to apply the **palette colors** consistently across the UI.
> * The **usability** of the submit survey screen could be improved. As a new user, it’s a bit unclear what's happening. I suggest adding more descriptive labels and instructions for the fields and submission process.
> * Why is there an **exclamation mark** after the submission date? Could you clarify its purpose? IMO, the text style is a bit inconsistent being black color instead of gray color like the other fields
> * Should we use a button to open the survey? IMO, probably the users could click the survey directly.
> 
> Let me know your thoughts on these points, and feel free to share any suggestions for improvement!

* I have added a new "Turn in late" button for survey that has already been past their due date.
* I have added Label text for textfields in submit-survey screen, this should improve its usability
* The exclamation mark after submission date was a typo from my part, it has been removed. I also have made its text black similar to other fields, also I'm still not sure if this is good or not
* I have removed the open button for the survey, now user can tap directly on the survey to open submit-survey screen
* Also

giolynx104 reviewed (approved):
@Ray276341 I struggle with UI design. Even after taking the UX/UI class, all I've really taken away is that if it feels good to you, that's what matters! 😂 The code was refined and I think it's ok to be merged.
